src/components/OptimizerForm.js

```diff
@@ -241,27 +232,30 @@ export default function OptimizerForm() {
 
   window.getVal = () => statDisplay
 
+
   const initialCharacter = useMemo(() => {
-    let characters = DB.getCharacters()
+    let characters = DB.getCharacters(); // retrieve instance localStore saved chars
     if (characters && characters.length > 0) {
-      let character = characters[0]
-      // let character = Utils.randomElement(characters)
-      console.log('Initial character', character)
+      let character = characters[0];
       lightConeSelectorChange(character.form.lightCone)
       setStatDisplay(character.form.statDisplay || 'base')
-
       return characterOptions.find(x => x.id == character.id)
-    } else {
-
     }
   }, []);
 
   const [selectedCharacter, setSelectedCharacter] = useState(() => initialCharacter);
   window.setSelectedCharacter = setSelectedCharacter
 
+  // TODO: refactor if/when view-routing/deep-linking implemented
+  // coming from char tab
+  const [selectedOptimizerCharacter, setSelectedOptimizerCharacter] = store(s => [s.selectedOptimizerCharacter, s.setSelectedOptimizerCharacter]);
+  if (selectedOptimizerCharacter && selectedOptimizerCharacter.id !== selectedCharacter.id) {
+    characterSelectorChange(selectedOptimizerCharacter.id);
+    setSelectedOptimizerCharacter(null);
+  }
```

Comment:
This might need to go into a useEffect or something, since its triggering the `Cannot update a component (`InternalFormItem`) while rendering a different component (`OptimizerForm`).` warning downstream

Reply:
Ah, makes sense.  Maybe also the reason i see multiple calls to rerender as well.  Let me spin on this a bit.  

Repro for above stack trace is just a char-select via dbl-click?  I could swear my console was clean before I opened the PR.

Reply:
PR updated, error no longer occurs.